lib/new-admin/login.js

```diff
@@ -1,48 +1,20 @@
-const crypto = require('crypto')
-
 const db = require('../db')
 
-function generateOTP (name) {
-  const otp = crypto.randomBytes(32).toString('hex')
-
-  const sql = 'insert into one_time_passes (token, name) values ($1, $2)'
-
-  return db.none(sql, [otp, name])
-    .then(() => otp)
-}
-
-function validateOTP (otp) {
-  const sql = `delete from one_time_passes
-    where token=$1
-    returning name, created < now() - interval '1 hour' as expired`
-
-  return db.one(sql, [otp])
-    .then(r => ({ success: !r.expired, expired: r.expired, name: r.name }))
-    .catch(() => ({ success: false, expired: false }))
-}
-
-function register (otp, ua, ip) {  
-  return validateOTP(otp)
-    .then(r => {
-      if (!r.success) return r
-
-      const token = crypto.randomBytes(32).toString('hex')
-      const sql = 'insert into user_tokens (token, name, user_agent, ip_address) values ($1, $2, $3, $4)'
-
-      return db.none(sql, [token, r.name, ua, ip])
-        .then(() => ({ success: true, token: token }))
-    })
-    .catch(() => ({ success: false, expired: false }))
+function checkUser (username) {
+  const sql = 'select * from users where username=$1'
+  return db.oneOrNone(sql, [username]).then(value => { return value.password }).catch(() => false)
 }
 
-function authenticate (token) {
-  const sql = 'select token from user_tokens where token=$1'
+function validateUser (username, password) {
+  const sql = 'select id, username from users where username=$1 and password=$2'
+  const sqlUpdateLastAccessed = 'update users set last_accessed = now() where username=$1'
 
-  return db.one(sql, [token]).then(() => true).catch(() => false)
+  return db.oneOrNone(sql, [username, password])
+    .then(user => { db.none(sqlUpdateLastAccessed, [user.username]); return user })
+    .catch(() => false)
 }
 
```

Comment:
This file is staring to have the same issue as our main Routes one. Can you look into separating the schema in different files? For now only do that for User related logic.

Reply:
schema.js should now have some modularization